app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt

```diff
@@ -553,6 +553,23 @@ class VaultRepositoryImpl(
         mutableTotpCodeResultFlow.tryEmit(totpCodeResult)
     }
 
+    override suspend fun unlockVaultWithAuthenticatorSyncKey(
+        userId: String,
+        decryptedAuthenticatorSyncKey: String,
+    ): VaultUnlockResult {
+        return unlockVaultForUser(
+            userId = userId,
+            initUserCryptoMethod = InitUserCryptoMethod.DecryptedKey(
+                decryptedUserKey = decryptedAuthenticatorSyncKey,
+            ),
+        )
+            .also {
+                if (it is VaultUnlockResult.Success) {
+                    deriveTemporaryPinProtectedUserKeyIfNecessary(userId = userId)
```

Comment:
Just curious...why is this necessary here?

Reply:
👍 Thanks!

Yeah when I was looking this over earlier I clearly didn't fully grok what was going on. From my reading, it seemed like it would be possible that this could have an impact but, looking it at it now, `deriveTemporaryPinProtectedUserKeyIfNecessary` has nothing to do with what we are doing. Admittedly, the `ifNecessary` suffix on the function name made me ✨feel✨  safe 😬 


I removed the `.also` call and updated test accordingly: 4230b7363313ec7502f94c7fdb7c51bd1836ac08